src/Routing/Filter/CookieEncryptionFilter.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Cake\Routing\Filter;
+
+use Cake\Core\Configure;
+use Cake\Event\Event;
+use Cake\Routing\DispatcherFilter;
+
+class CookieEncryptionFilter extends DispatcherFilter
+{
+
+    const CONFIG_KEY = 'Cookie';
+
+    /**
+     * Cookies should be decrypted after jars have been set.
+     *
+     * @var int
+     */
+    protected $_priority = 25;
+
+    /**
+     *
+     * @var string
+     */
+    protected $_cookieEncrypterClassName = 'Cake\Network\Cookie\CookieEncrypter';
+
+    /**
+     *
+     * @var \Cake\Network\Cookie\CookieEncrypter
+     */
+    protected $_encrypter;
+
+    /**
+     *
+     * @param \Cake\Event\Event $event
+     * @return void
+     */
+    public function beforeDispatch(Event $event)
+    {
+        $request = $event->data['request'];
+        $this->_decryptCookies($request);
```

Comment:
@ADmad I'm afraid that's the case with encryption being made in a filter. Check previous commits where I proposed lazy cookie decryption. Different solution has been mentioned or implemented here. It's just a matter of deciding which is the best one.